File: backend/app/api/features.py
from fastapi import APIRouter, HTTPException

from app.features import (
    FeatureExtractionRequest,
    FeatureExtractionResponse,
    extract_windowed_magnetometer_features,
)

router = APIRouter()


@router.post(
    "/features/vector-magnetometer/extract",
    response_model=FeatureExtractionResponse,
)
def extract_vector_magnetometer_features(
    request: FeatureExtractionRequest,
) -> FeatureExtractionResponse:
    try:
        return extract_windowed_magnetometer_features(request)
    except ValueError as exc:
        raise HTTPException(status_code=422, detail=str(exc)) from exc

File: frontend/src/api/features.ts
import type {
  FeatureExtractionRequest,
  FeatureExtractionResponse,
} from "../types/features";
import { requestJson } from "./client";

export function extractVectorMagnetometerFeatures(
  request: FeatureExtractionRequest,
  signal?: AbortSignal,
): Promise<FeatureExtractionResponse> {
  return requestJson<FeatureExtractionResponse>(
    "/api/features/vector-magnetometer/extract",
    {
      method: "POST",
      body: request,
      signal,
    },
  );
}
